include/mlir-tcp/Dialect/IR/TcpTypes.td

```diff
@@ -16,11 +16,19 @@ include "mlir/IR/DialectBase.td"
 include "mlir-tcp/Dialect/IR/TcpBase.td"
 
 //===----------------------------------------------------------------------===//
-// Tcp Type Definitions.
+// Tcp Type Definitions
+//===----------------------------------------------------------------------===//
+
+def Tcp_IndexArrayType : Tcp_Type<"IndexArray", "index_array"> {
+    let summary = "IndexArray TCP type, to holds a list of index builtin type to represent shape";
+}
+
```

Comment:
very minor nit: It might be better to have the basic types, `Tcp_Scalar`, `Tcp_Tensor` before we define `IndexArrayType`. The same would apply to the `Tcp_QuantizedType` as well. So, may be we can do it later.

Reply:
`Tcp_Scalar` depends on `Tcp_QuantizedInt` so it has to follow quantized types. However I've moved Tcp_IndexArrayType to the end.